### DISCUSSION TITLE
Four points of feedback from my Copilot flight:

### DISCUSSION THREAD

**Discussion by JasonHoku** [community/community#6674] [Copilot Conversations]
Hello everyone, I want to start this feedback with a thank you to everyone for all coming together to work on this revolutionary technology. I really am so grateful to be part of this group and to be able to test and offer ideas on this truly game changing utility.

### 1. Voting & Reporting
 Firstly, in my opinion the most important feature would be a feedback/report/downvote/remove this solution button. There seems to be a lot of depreciated, outdated and bad code and maybe some form of npm/package manager intelligence for package version specific syntax would be the next level solution to this issue.

### 2.  Sense Wrapper
Secondly another feature that I think would improve performance and workflow would be a string syntax wrapper for only sensing between two comments to prevent needing to create a new file to generate code out of context of the entire working file. Both the documentation says this is the correct way and from my testing Copilot seem to really struggle in larger files so a comment wrapper solution sounds like a nice way to not have to do so much tab swapping and keyboard acrobatics.

### 3. More Keyboard Hotkeys
Thirdly, more keyboard hotkeys for swapping between, like next and back, (and ideally additionally removing) the more solutions list would be amazing.
 
 
### 4. Better Issue/Feature Tracking
And lastly I seem to have trouble finding a list of features that are being worked on, or even discussions about issues. I have tried searching for relevant terms to my ideas and problems in the GitHub Copilot feedback forum but didn't find any relevant results the way I have in other GitHub repos through the issues tab, which we don't really seem to have here other than the private forum. So either the forum might need improving for the search and issue reports or perhaps we need another place to list current issues, solved issues, and feature requests.

Again, thank you all for the work here, I hope some of these thoughts are helpful or are even reaching the right audience. If anyone thinks this belongs anywhere else or in any other form, definitely let me know.

**Comment by SmallLemur** [comment]
Hi Jason,

Thank you for your detailed feedback, it is definitely reaching the right audience. We are still in the process of analysing and synthesising all the feedbacks we received !

Fabien

**Reply by SmallLemur** [reply]
@JasonHoku following up on your comment, Specifically on point #2 do you mean that the option of seeing the context that Copilot is seeing would be helpful ?

**Reply by JasonHoku** [reply]
Yes. To clarify, I believe that Copilot uses the entire current file to determine it's predictions, and additional control could be introduced by a wrapper consisting of syntax like: 


//NewCopilot

//Do something without making a new file and without sensing the current file

//EndCopilot

It feels like a quicker workflow than making a fresh file, and then selecting the language in vsc, writing your comments, then finally copy pasting into your desired file.

I hope that helps! 

Thank you again, so much, for the awesome work and utility!!

**Reply by Symbolk** [reply]
It looks like a ChatOps style, where you call/summon Copilot with a sensable open command, and then put a away with a close command.

But, these commands or tags should not exist in the final code comments, so a better way to control is either with mouse/keyshortcuts, or automatically remove all commands after saving.

**Comment by Symbolk** [comment]
+1, a place to better organize the features [todo, doing, done] uniformly would be really nice!
